src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1187,6 +1188,33 @@ namespace winrt::TerminalApp::implementation
         return false;
     }
 
+    // Method Description:
+    // - Helper to manually exit "zoom" when certain actions take place.
+    //   Anything that modifies the state of the pane tree should probably
+    //   un-zoom the focused pane first, so that the user can see the full pane
+    //   tree again. These actions include:
+    //   * Splitting a new pane
+    //   * Closing a pane
+    //   * Moving focus between panes
+    //   * Resizing a pane
```

Comment:
All of this feels weird to me. Like, I guess it's correct? If somebody tries to modify the pane structure, we should UnZoom, and _then_ do the action.

At the same time, this is how I feel about each one:
- split: ✔
- close: ✔
- move focus:
   - maybe it should just not be allowed :/
   - or switch panes but make that other pane be zoomed in?
- resize:
   - maybe it should not be allowed :/

What do others do? I might be the only one thinking this is weird too, in which case, just ignore me (hence not blocking)

Reply:
> move focus:
> * or switch panes but make that other pane be zoomed in?


holy crap I love that so much

Reply:
no promises as part of this PR. I'm gonna try it now, but if I can't get it trivially, I'm filing a follow-up cause that's an awesome idea

Reply:
Okay, this is _not_ trivial. When we move the focus to another pane, it takes a dispatcher loop to be able to mark the newly-focused pane as the "active" one. So if we do it all in the `moveFocus` handler, then when we try to zoom in on the active pane, it's still _technically_ the current pane, not the new one.

I got it to _sorta_ work with

```c++
    void TerminalPage::_MoveFocus(const Direction& direction)
    {
        if (auto index{ _GetFocusedTabIndex() })
        {
            auto focusedTab{ _GetStrongTabImpl(*index) };
            const bool wasZoomed = focusedTab->IsZoomed();
            _UnZoomIfNeeded();
            focusedTab->NavigateFocus(direction);
            if (wasZoomed)
            {
                _tabView.Dispatcher().RunAsync(CoreDispatcherPriority::Low, [this, focusedTab]() {
                    focusedTab->ToggleZoom();

                    // Update the selected tab, to trigger us to re-add the tab's GetRootElement to the UI tree
                    _UpdatedSelectedTab(_tabView.SelectedIndex());
                });
            }
        }
    }
```

But that causes a frame where we re-attach the zoomed-out UI, then go back to the zoomed-in UI (zoomed to the new pane). Unfortunately, that forces 2 resizes (resize the current pane smaller, then resize the new pane bigger), and those resizes in debug are fairly laggy.

I'm gonna punt this to a follow up, because it's a cool idea, but not one for right now.